### PR TITLE
Commands

### PR DESCRIPTION
**Pull request by Jonahss**

### CODE REVIEW

**Comment by Jonahss** [comment]
Do not merge yet.
Seems like somehow my emulator crash more often after this change...... will investigate.

File: lib/commands/general.js [hunk 1]
```diff
@@ -148,6 +148,29 @@ commands.getStrings = async function (language) {
   return this.apkStrings[language];
 };
 
+commands.startActivity = function (appPackage, appActivity) {
```
Comment (jlipps) [inline lib/commands/general.js]: this isn't async?

File: lib/commands/general.js [hunk 2]
```diff
@@ -148,6 +148,29 @@ commands.getStrings = async function (language) {
...
+
+commands.reset = async function () {
+  if (this.opts.fastReset) {
+    log.info("Running fast reset (stop and clear)");
+    return this.adb.stopAndClear(this.opts.appPackage);
+  }
+
+  log.info("Running old fashion reset (reinstall)");
+  await this.adb.uninstallApk(this.opts.appPackage);
+  return androidHelpers.installApkRemotely(this.adb, this.opts.app, this.opts.appPackage, this.opts.fastReset);
```
Comment (jlipps) [inline lib/commands/general.js]: we're doing `return await` now

**Comment by jlipps** [comment]
one comment, then LGTM (pending resolution of your crash comment)

File: lib/commands/general.js [hunk 3]
```diff
@@ -148,6 +148,29 @@ commands.getStrings = async function (language) {
...
+    activity: appActivity,
+    action: this.opts.intentAction,
+    category: this.opts.intentCategory,
+    flags: this.opts.intentFlags,
+    optionalIntentArguments: this.opts.optionalIntentArguments,
+    stopApp: this.opts.stopAppOnReset || !this.opts.dontStopAppOnReset,
+  });
+};
+
+commands.reset = async function () {
```
Comment (moizjv) [inline lib/commands/general.js]: https://github.com/appium/appium/blob/master/lib/appium.js#L380 we need this reset

File: lib/commands/general.js [hunk 4]
```diff
@@ -148,6 +148,32 @@ commands.getStrings = async function (language) {
...
+  if (this.opts.fastReset) {
+    log.info("Running fast reset (stop and clear)");
+    return this.adb.stopAndClear(this.opts.appPackage);
```
Comment (moizjv) [inline lib/commands/general.js]: in this case dont we need to `await this.startAUT()` ?

File: lib/commands/general.js [hunk 5]
```diff
@@ -148,6 +148,32 @@ commands.getStrings = async function (language) {
   return this.apkStrings[language];
 };
 
+commands.startActivity = async function (appPackage, appActivity) {
+  return this.adb.startApp({
```
Comment (jlipps) [inline lib/commands/general.js]: what does this return? if nothing, we don't need it.

also, we need an `await` here

File: lib/commands/general.js (hunk 4 above)
Comment (jlipps) [inline lib/commands/general.js]: correct; we also need to `await this.adb.stopAndClear`

Comment (jlipps) [inline lib/commands/general.js]: sounds like we should just do away with the `return` here and `await` instead

File: lib/driver.js [hunk 6]
```diff
@@ -43,29 +43,39 @@ class AndroidDriver extends BaseDriver {
...
+                         category: "android.intent.category.LAUNCHER",
+                         flags: "0x10200000",
+                         disableAndroidWatchers: false,
+                         tmpDir: await tempDir.staticDir(),
+                         fullReset: false};
+      _.defaults(this.opts, defaultOpts);
+      if (!this.opts.javaVersion) {
+        this.opts.javaVersion = await helpers.getJavaVersion();
+      }
+      // Add condition for chrome driver
```
Comment (jlipps) [inline lib/driver.js]: fyi, this is going to conflict with my PR as well, guess we'll get to see whose is merged first :-)

File: lib/commands/general.js (hunk 4 above)
Comment (Jonahss) [inline lib/commands/general.js]: ok

File: lib/driver.js (hunk 6 above)
Comment (Jonahss) [inline lib/driver.js]: I'll handle conflicts :)

Comment (jlipps) [inline lib/driver.js]: you mean by merging this that I'll handle conflicts I guess :-)

Comment (jlipps) [inline lib/driver.js]: conflicts handled!

Comment (Jonahss) [inline lib/driver.js]: I meant I'd be fine merging the other PR manually for you :)

Comment (jlipps) [inline lib/driver.js]: ah ok thanks :-)

**Pull request merged**